utils.test.js

```diff
@@ -901,54 +901,101 @@ test("parse go version data", async () => {
+        value: "./test/Cargo.lock"
+      }
+    ]
   });
-  dep_list = await parseCargoData(
-    readFileSync("./test/data/Cargom.lock", { encoding: "utf-8" })
-  );
+
+  dep_list = await parseCargoData("./test/data/Cargom.lock");
   expect(dep_list.length).toEqual(243);
   expect(dep_list[0]).toEqual({
+    type: "library",
     group: "",
+    "bom-ref": "pkg:cargo/actix-codec@0.3.0",
+    purl: "pkg:cargo/actix-codec@0.3.0",
     name: "actix-codec",
     version: "0.3.0",
-    _integrity:
-      "sha384-78d1833b3838dbe990df0f1f87baf640cf6146e898166afe401839d1b001e570"
+    hashes: [
+      {
+        alg: "SHA-384",
+        content:
+          "78d1833b3838dbe990df0f1f87baf640cf6146e898166afe401839d1b001e570"
+      }
+    ],
+    evidence: {
+      identity: {
+        field: "purl",
+        confidence: 0.6,
+        methods: [
+          {
+            technique: "manifest-analysis",
+            confidence: 0.6,
+            value: "./test/data/Cargom.lock"
+          }
+        ]
+      }
+    },
+    properties: [
+      {
+        name: "SrcFile",
+        value: "./test/data/Cargom.lock"
+      }
+    ]
   });
+
+  // The base64 package does not have an associated checksum. Make sure the
+  // function does not accidentally insert an undefined hashsum value.
+  const base64Package = dep_list.find((pkg) => pkg.name === "base64");
+  expect(base64Package).not.toContain("hashes");
+});
+
+test("parse cargo lock simple component representation", async () => {
+  // If asking for a simple representation, we should skip any extended attributes.
+  const componentList = await parseCargoData("./test/Cargo.lock");
+  const firstPackage = componentList[0];
+  expect(firstPackage).not.toContain("evidence");
 });
 
 test("parse cargo lock lists last package", async () => {
   // The implementation procedurally fills an object with the package
   // information line-by-line, considering a package's information "complete"
   // when the next package is found. This risks missing the last package in
   // the file, so this test case makes sure it is still found.
-  const cargoFileContent = `# This file is automatically @generated by Cargo.
-# It is not intended for manual editing.
-version = 3
-
-[[package]]
-name = "package1"
-version = "1.21.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "8a30b2e23b9e17a9f90641c7ab1549cd9b44f296d3ccbf309d2863cfe398a0cb"
-
-[[package]]
-name = "package2"
-version = "2.10.0"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "8a30b2e23b9e17a9f90641c7ab1549cd9b44f296d3ccbf309d2863cfe398a0cb"
-`;
-  const componentList = await parseCargoData(cargoFileContent);
-  expect(componentList.length).toBeGreaterThan(1);
-  expect(componentList.find((pkg) => pkg.name === "package2")).toBeTruthy();
+  const componentList = await parseCargoData("./test/data/Cargom.lock");
+  expect(componentList.find((pkg) => pkg.name === "yaml-rust")).toBeTruthy();
```

Comment:
For test cases describing specific cases with a minimal reproducible input, I usually prefer to be able to have the input be inline so as to be easy to get an overview of it, and a corresponding description.

Because `parseCargoData` now needs a file, such situations would require temporary files and logistics for that is not currently available. For that reason, a larger test file containing the same requirement for the test was picked.